torch_xla/csrc/tensor_methods.cpp

```diff
@@ -113,6 +113,28 @@ void CheckDimensionSize(const XLATensor& t, xla::int64 dim,
       << " (while checking arguments for " << tag << ")";
 }
 
+// Resizes and / or checks whether a list is of the given size. The list is only
+// resized if its size is 1. If it's empty, it's replaced with the provided
+// default first.
+std::vector<xla::int64> CheckIntList(
+    tensorflow::gtl::ArraySlice<const xla::int64> list, size_t length,
+    const std::string& name, std::vector<xla::int64> def = {}) {
+  std::vector<xla::int64> result;
+  if (list.empty()) {
+    result = std::move(def);
+  } else {
+    result = xla::util::ToVector<xla::int64>(list);
+  }
+  if (result.size() == 1 && length > 1) {
+    result.resize(length, result[0]);
+    return result;
+  }
+  XLA_CHECK_EQ(result.size(), length)
```

Comment:
You do not need to repeat streaming the two values, as those are part of the CHECK message. You stream the name and some other textual context and you are done.

Reply:
Done.